File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        inOrd(root,res);
        return res;
    }
    
    void inOrd(TreeNode root, List<Integer> res){
        // Base Case:
        if(root == null) return;
        
        // Recursive Call:
        // First, Left Sub Tree:
        inOrd(root.left, res);
        
        // Second, Printing the Root Node:
        res.add(root.val);
        
        // Third, Right Sub Tree:
        inOrd(root.right, res);
    }
}

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        self.f(root, res)
        return res
    
    def f(self, root: Optional[TreeNode], res: List[int]):
        # Base Case:
        if(root == None): return None
        
        # Recursive Call:
        self.f(root.left, res)
        res.append(root.val)
        self.f(root.right, res)
        